cozy/synthesis/high_level_interface.py

```diff
@@ -99,7 +100,8 @@ def improve_implementation(
         impl              : Implementation,
         timeout           : datetime.timedelta = datetime.timedelta(seconds=60),
         progress_callback : Callable[[Implementation], Any] = None,
-        improve_count     : Value = None) -> Implementation:
+        improve_count     : Value = None,
+        save              : str = None) -> Implementation:
```

Comment:
The name `save` is a bit terse. I would suggest `save_file_name` instead. It would also be nice to put a sentence or two in the procedure docstring about what this parameter does.